internal/agent/logs.go

```diff
@@ -0,0 +1,271 @@
+package agent
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/kairos-io/kairos-agent/v2/pkg/config"
+	"github.com/kairos-io/kairos-agent/v2/pkg/constants"
+	v1 "github.com/kairos-io/kairos-agent/v2/pkg/types/v1"
+	fsutils "github.com/kairos-io/kairos-agent/v2/pkg/utils/fs"
+	"github.com/kairos-io/kairos-sdk/collector"
+	"github.com/kairos-io/kairos-sdk/types"
+)
+
+// LogsResult represents the collected logs
+type LogsResult struct {
+	Journal map[string][]byte `yaml:"-"`
+	Files   map[string][]byte `yaml:"-"`
+}
+
+// LogsCollector handles the collection of logs from various sources
+type LogsCollector struct {
+	config *config.Config
+}
+
+// NewLogsCollector creates a new LogsCollector instance
+func NewLogsCollector(cfg *config.Config) *LogsCollector {
+	return &LogsCollector{
+		config: cfg,
+	}
+}
+
+func defaultLogsConfig() *config.LogsConfig {
+	return &config.LogsConfig{
+		Journal: []string{
+			"kairos-agent",
+			"kairos-installer",
+			"kairos-webui",
+			"cos-setup-boot",
+			"cos-setup-fs",
+			"cos-setup-network",
+			"cos-setup-reconcile",
+			"k3s",
+			"k3s-agent",
+			"k0scontroller",
+			"k0sworker",
+		},
+		Files: []string{
+			"/var/log/kairos/*.log",
+			"/run/immucore/*.log",
```

Comment:
Would these be enough logs in an openrc distro?

Reply:
k3s logs to /var/log/k3s.log I think? so maybe everything under /var/log would be useful for openrc